src/screens/smart-screen/smart-screen.tsx

```diff
@@ -0,0 +1,216 @@
+import React from 'react';
+import { View } from 'react-native';
+import { smartConnect } from '../../core/utils/smart-connect';
+import { connect } from 'react-redux';
+import { Widgets } from '../../components/widgets/widgets';
+import { fetchScreenData } from '../../redux/ui/screens/data/actions';
+import { IScreenContext } from '../../components/widgets/types';
+import { IReduxState } from '../../redux/state';
+import { IScreenData, IScreenDatas } from '../../redux/ui/screens/data/state';
+import { withdraw, claimRewardNoInput } from '../../redux/wallets/actions/pos-actions';
+import { getScreenDataKey } from '../../redux/ui/screens/data/reducer';
+import { getSelectedAccount, getSelectedWallet } from '../../redux/wallets/selectors';
+import { getChainId } from '../../redux/preferences/selectors';
+import { IAccountState } from '../../redux/wallets/state';
+import { getAccountStats } from '../../redux/ui/stats/selectors';
+import { AccountStats } from '../../core/blockchain/types/stats';
+import { SkeletonPlaceholder } from '../../components/skeleton-placeholder/skeleton-placeholder';
+import stylesProvider from './styles';
+import { IThemeProps, withTheme } from '../../core/theme/with-theme';
+import { ErrorWidget } from '../../components/widgets/components/error-widget/error-widget';
+import { translate } from '../../core/i18n';
+
+interface IExternalProps {
+    context: IScreenContext;
+}
+
+const mapStateToProps = (state: IReduxState) => {
+    const account = getSelectedAccount(state);
+    const chainId = account && getChainId(state, account.blockchain);
+
+    return {
```

Comment:
well this is not scalable at all :) let's do something generic. you should return here a `screenData` key that has the value of `state.ui.screens.data[ownProps.context.screen]` this way you return only what you need in the screen. Of course you need to handle some errors cases.

Reply:
this way you can remove `getScreenData` method :) i think it won't be needed anymore